src/OpenSpaceToolkit/Astrodynamics/Flight/System/Dynamics/SatelliteDynamics.cpp

```diff
@@ -160,63 +165,47 @@ void SatelliteDynamics::DynamicalEquations(const Dynamics::StateVector& x, Dynam
         totalGravitationalAcceleration_SI += gravitationalAcceleration.inFrame(gcrfSPtr_, currentInstant).getValue();
     }
 
-    // Integrate position and velocity states
+    dxdt[3] += totalGravitationalAcceleration_SI[0];
+    dxdt[4] += totalGravitationalAcceleration_SI[1];
+    dxdt[5] += totalGravitationalAcceleration_SI[2];
+
+    // DRAG
+
+    const Real mass = satelliteSystem_.getMass().inKilograms();
+    const Real dragCoefficient = satelliteSystem_.getDragCoefficient();
+    const Real surfaceArea = satelliteSystem_.getCrossSectionalSurfaceArea();
+
+    for (const auto& objectName : environment_.getObjectNames())
+    {
+        // TBI: currently only defined for Earth
+        if (environment_.accessCelestialObjectWithName(objectName)->accessAtmosphericModel() != nullptr)
+        {
+            const Real atmosphericDensity = environment_.accessCelestialObjectWithName(objectName)
+                                                ->getAtmosphericDensityAt(currentPosition)
+                                                .getValue();
+
+            // [TBI]: Define in Physics celestial body
+            const Vector3d earthAngularVelocity = {0, 0, 7.2921159e-5};  // rad/s
```

Comment:
I think it's defined in the Earth model already?

Reply:
^ @kyle-cochran That's what we were mentioning in our tag-up earlier ;)

Reply:
Aha! Nice!